sql: TestPGPreparedQuery fails on release-2.1, but not on TC

I have no idea what's going on, but this test seems to fail constantly for me and several others on `release-2.1`. Except on TC it seems to pass reliably.

```
--- FAIL: TestPGPreparedQuery (2.14s)
    --- FAIL: TestPGPreparedQuery/prepare (0.83s)
        --- FAIL: TestPGPreparedQuery/prepare/SELECT_CASE_40+2_WHEN_42_THEN_51_ELSE_$1::INT_END (0.00s)
            --- FAIL: TestPGPreparedQuery/prepare/SELECT_CASE_40+2_WHEN_42_THEN_51_ELSE_$1::INT_END/0 (0.00s)
                pgwire_test.go:1013: SELECT CASE 40+2 WHEN 42 THEN 51 ELSE $1::INT END: []: expected error: sql: statement expects 1 inputs; got 0, got sql: expected 1 arguments, got 0
```

@knz would you mind finding a victim for this one?

Radu knows where to look on this one.

Otherwise Matt

Andrei Matei <notifications@github.com> schreef op 26 maart 2019 22:36:55 CET:
>I have no idea what's going on, but this test seems to fail constantly
>for me and several others on `release-2.1`. Except on TC it seems to
>pass reliably.
>
>```
>--- FAIL: TestPGPreparedQuery (2.14s)
>    --- FAIL: TestPGPreparedQuery/prepare (0.83s)
>--- FAIL:
>TestPGPreparedQuery/prepare/SELECT_CASE_40+2_WHEN_42_THEN_51_ELSE_$1::INT_END
>(0.00s)
>--- FAIL:
>TestPGPreparedQuery/prepare/SELECT_CASE_40+2_WHEN_42_THEN_51_ELSE_$1::INT_END/0
>(0.00s)
>pgwire_test.go:1013: SELECT CASE 40+2 WHEN 42 THEN 51 ELSE $1::INT END:
>[]: expected error: sql: statement expects 1 inputs; got 0, got sql:
>expected 1 arguments, got 0
>```
>
>@knz would you mind finding a victim for this one?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/cockroachdb/cockroach/issues/36184

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

--- a/pkg/sql/pgwire/pgwire_test.go
+++ b/pkg/sql/pgwire/pgwire_test.go
@@ -57,10 +57,7 @@ import (
 // https://golang.org/cl/38533 and https://golang.org/cl/91115 changed the
 // validation message.
 func wrongArgCountString(want, got int) string {
-	if strings.HasPrefix(runtime.Version(), "go1.10") {
-		return fmt.Sprintf("sql: expected %d arguments, got %d", want, got)
-	}
-	return fmt.Sprintf("sql: statement expects %d inputs; got %d", want, got)
+	return fmt.Sprintf("sql: expected %d arguments, got %d", want, got)
 }
 
 func trivialQuery(pgURL url.URL) error {

